48_ima_incremental_attestation.md

```diff
@@ -0,0 +1,307 @@
+not respond with the partial log but the entire IMA log. This situation
+is indicated by the agent not returning the number of the first IMA log
+entry it returned, thus the verifier will assume the log starts at
+the first entry.
+
+From the above it can be seen that the benefits of reduction of network
+bandwidth consumption and CPU cycle consumption are only going to take
+effect when updated verifiers and updated agents are being used.
+
+Upon restart of the verifier, it will request the IMA log from the
+first entry (entire log) when resuming communication with a particular
+agent. Therefore, startup of the IMA verification will be as costly
+(in terms of resource consumption) as it was before this enhancement.
+
+The per-agent state of the PCRs and the next-to-request IMA log entry
+will be maintained in the verifier's memory and be internally accessible
+using the unique ID of an agent (indexed by agent Id) as a
+search/selection criterion.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this enhancement is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+enhancement ecosystem.
+
+How will security be reviewed and by whom?
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Many of the details are already given above.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+The PR implementing this proposal will add
+- unit tests for newly added functions
+- integration tests for testing incremental attestation
+
+### Upgrade / Downgrade Strategy
+
```

Comment:
I would mention your above points about older verifiers and agents being able to coexist and falling back to previous behaviour. Also mention that there won't be any database schema upgrades as part of this.